File: Data_Structure/DS_Using_Python/Binary_Tree.py
class BinaryTree:
	def __init__(self, data):
		self.data = data
		self.left = None
		self.right = None
		
	def inOrder(self, tree):
		if tree is None:
			return
		self.inOrder(tree.left)
		print(tree.data, end=' ')
		self.inOrder(tree.right)
		
	def preOrder(self, tree):
		if tree is None:
			return
		print(tree.data, end=' ')
		self.inOrder(tree.left)
		self.inOrder(tree.right)
		
	def postOrder(self, tree):
		if tree is None:
			return
		self.inOrder(tree.left)
		self.inOrder(tree.right)
		print(tree.data, end=' ')
		
	def insertLeft(self, newNode):
		if self.left is None:
			self.left = BinaryTree(newNode)
		else:
			tree = BinaryTree(newNode)
			tree.left = self.left
			self.left = tree
		
	def insertRight(self, newNode):
		if self.right is None:
			self.right = BinaryTree(newNode)
		else:
			tree = BinaryTree(newNode)
			tree.right = self.right
			self.right = tree
		
root = BinaryTree(4)
root.insertLeft(8)
root.insertRight(5)
root.insertLeft(9)
root.insertRight(85)
root.insertLeft(29)
root.insertRight(45)


print('Inorder Traversal:')
root.inOrder(root)

print('\n')
print('Preorder Traversal:')
root.preOrder(root)

print('\n')
print('Postorder Traversal:')
root.postOrder(root)

File: Data_Structure/Trees/Binary_Tree.c
#include<stdio.h>

struct node
{
    int data;
    struct node* left;
    struct node* right;
};

struct node* createNode(int item)
{
    struct node* newNode = (struct node*)malloc(sizeof(struct node));
    newNode -> data = item;
    newNode -> left = NULL;
    newNode -> right = NULL;
    return newNode;
}

void inOrder(struct node* root)
{
    if(root == NULL)
    {
        return;
    }
    inOrder(root -> left);
    printf("%d ", root -> data);
    inOrder(root -> right);
}

void preOrder(struct node* root)
{
    if(root == NULL)
    {
        return;
    }
    printf("%d ", root -> data);
    preOrder(root -> left);
    preOrder(root -> right);
}

void postOrder(struct node* root)
{
    if(root == NULL)
    {
        return;
    }
    postOrder(root -> left);
    postOrder(root -> right);
    printf("%d ", root -> data);
}

struct node* insertLeft(struct node* root, int item)
{
    root -> left = createNode(item);
    return root -> left;
}

struct node* insertRight(struct node* root, int item)
{
    root -> right = createNode(item);
    return root -> right;
}

int main()
{
    struct node* root = createNode(2);
    insertLeft(root, 5);
    insertRight(root, 7);
    insertLeft(root -> left, 8);
    
    printf("Inorder Traversal:\n");
    inOrder(root);
    
    printf("\n\n");
    printf("Preorder Traversal:\n");
    preOrder(root);
    
    printf("\n\n");
    printf("Postorder Traversal:\n");
    postOrder(root);
    return 0;
}